When exporting Markdown, encountered nil pointer dereference

Version: v2.9.10

When exporting Markdown, encountered nil pointer dereference, example:

```
$ ./nuclei -silent -t ~/nuclei-templates/dns/dns-waf-detect.yaml -duc -me . -u shellcodes.org
[dns-waf-detect:cloudflare] [dns] [info] shellcodes.org
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x127a1a8]
```

The reason is that some template files do not have 'reference' field.

--- a/v2/cmd/integration-test/template-path.go
+++ b/v2/cmd/integration-test/template-path.go
@@ -20,7 +20,7 @@ var templatesPathTestCases = []TestCaseInfo{
 	//relative path
 	{Path: "dns/dns-saas-service-detection.yaml", TestCase: &relativePathTemplateTest{}},
 	//absolute path
-	{Path: fmt.Sprintf("%v/dns/cname-fingerprint.yaml", getTemplatePath()), TestCase: &absolutePathTemplateTest{}},
+	{Path: fmt.Sprintf("%v/dns/dns-saas-service-detection.yaml", getTemplatePath()), TestCase: &absolutePathTemplateTest{}},
 }
 
 type cwdTemplateTest struct{}

--- a/v2/cmd/integration-test/template-path.go
+++ b/v2/cmd/integration-test/template-path.go
@@ -18,7 +18,7 @@ var templatesPathTestCases = []TestCaseInfo{
 	//cwd
 	{Path: "./dns/cname-fingerprint.yaml", TestCase: &cwdTemplateTest{}},
 	//relative path
-	{Path: "dns/cname-fingerprint.yaml", TestCase: &relativePathTemplateTest{}},
+	{Path: "dns/dns-saas-service-detection.yaml", TestCase: &relativePathTemplateTest{}},
 	//absolute path
 	{Path: fmt.Sprintf("%v/dns/cname-fingerprint.yaml", getTemplatePath()), TestCase: &absolutePathTemplateTest{}},
 }

--- a/v2/pkg/reporting/format/format_utils.go
+++ b/v2/pkg/reporting/format/format_utils.go
@@ -113,7 +113,7 @@ func CreateReportDescription(event *output.ResultEvent, formatter ResultFormatte
 	}
 
 	reference := event.Info.Reference
-	if !reference.IsEmpty() {
+	if reference != nil && !reference.IsEmpty() {
 		builder.WriteString("\nReferences: \n")
 
 		referenceSlice := reference.ToSlice()

